scripts/calc_percent_coverage.py

```diff
@@ -1,70 +1,113 @@
+
+
+def parse_args(args: list[str]) -> argparse.Namespace:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Calculate percent coverage.")
 
-#### FUNCTIONS #####
-def getOptions(args=sys.argv[1:]):
-    parser = argparse.ArgumentParser(description="Parses command.")
-    
-    parser.add_argument("--sample_name",  help= "")
-    parser.add_argument('--fasta_file', help = '')
-    parser.add_argument('--reference_file', help = '')
+    parser.add_argument(
+        "--log_level",
+        help="the level to log at",
+        choices=["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"],
+        default="INFO",
+    )
+    parser.add_argument("--sample_name", help="sample name")
+    parser.add_argument("--fasta_file", help="fasta file")
+    parser.add_argument("--reference_file", help="reference file")
     options = parser.parse_args(args)
+
     return options
 
 
-def calculate_percent_coverage(sample_name, fasta_file, reference_file):
+def setup_logging(log_level: str) -> None:
+    """Set up logging."""
+    log_format = "[%(asctime)s] %(levelname)s:%(name)s:%(message)s"
+    logging.basicConfig(
+        level=log_level,
+        stream=sys.stdout,
+        format=log_format,
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+
+def calculate_percent_coverage(
+    sample_name: str, fasta_file: str, reference_file: str
+) -> None:
+    """Calculate percent coverage."""
     # first check that there is only one sequence in the fasta files
     num_records = 0
-    with open(fasta_file, 'r') as fasta_handle:
+    with open(fasta_file, "r") as fasta_handle:
         inside_text = fasta_handle.read()
-    for item in re.finditer('>', inside_text):
+    for _ in re.finditer(">", inside_text):
         num_records = num_records + 1
 
     # calculate the percent coverage
-    ref_genome = SeqIO.read(reference_file, 'fasta')
+    ref_genome = SeqIO.read(reference_file, "fasta")
     ref_genome_length = len(ref_genome.seq)
+
+    Ns = 0
+    coverage = 0.0
+    aligned_bases = 0
+    num_non_ambiguous_bases = 0
     if num_records == 1:
-        record = SeqIO.read(fasta_file, 'fasta')
+        record = SeqIO.read(fasta_file, "fasta")
         aligned_bases = len(record.seq)
         if aligned_bases == 0:
             Ns = ref_genome_length
             num_non_ambiguous_bases = 0
             coverage = 0
         else:
-            uncorrected_Ns = record.seq.count('N')
+            uncorrected_Ns = record.seq.count("N")
             Ns = (ref_genome_length - aligned_bases) + uncorrected_Ns
             num_non_ambiguous_bases = aligned_bases - uncorrected_Ns
-            coverage = round((1-(Ns/ref_genome_length)) * 100, 2)
+            coverage = round((1 - (Ns / ref_genome_length)) * 100, 2)
     else:
-        aligned_bases = np.NaN
-        Ns = np.NaN
-        num_non_ambiguous_bases = np.NaN
-        coverage = np.NaN
+        aligned_bases = np.NaN  # type: ignore
+        Ns = np.NaN  # type: ignore
```

Comment:
I'm not sure how to handle the type issues where we assign `some_variable = np.NaN`. I get an error from mypy `error: Incompatible types in assignment (expression has type "float", variable has type "int")`.